app-backend/api/src/main/scala/utils/QueryParameters.scala

```diff
@@ -69,4 +72,24 @@ trait QueryParametersCommon extends QueryParameterDeserializers {
   def shapeQueryParams = (
     orgQueryParams & userQueryParameters & timestampQueryParameters
   ).as(ShapeQueryParameters.apply _)
+
+  def searchQueryParams = parameters(
+    'searchPattern.as[String].?
+  ).as(SearchQueryParameters.apply _)
+
+  def toolParameters = parameters(
```

Comment:
I think the tool parameters should be moved to a tool specific file next to the tool routes instead of being put into the util file.